tests/functional_test_base.go

```diff
@@ -119,6 +119,11 @@ func (s *FunctionalTestBase) setupSuite(defaultClusterConfigFile string, options
 
 	clusterConfig, err := GetTestClusterConfig(defaultClusterConfigFile)
 	s.Require().NoError(err)
+	if s.dynamicConfigOverrides == nil {
+		s.dynamicConfigOverrides = make(map[dynamicconfig.Key]interface{})
+	}
+	s.dynamicConfigOverrides[dynamicconfig.HistoryScannerEnabled] = false
+	s.dynamicConfigOverrides[dynamicconfig.TaskQueueScannerEnabled] = false
 	clusterConfig.DynamicConfigOverrides = s.dynamicConfigOverrides
```

Comment:
also, just a thought, what if you wrote this like this, to avoid mutating and to use a map literal which is a little nicer to read:
```suggestion
	clusterConfig.DynamicConfigOverrides = util.CloneMapNonNil(s.dynamicConfigOverrides)
	maps.Copy(clusterConfig.DynamicConfigOverrides, map[dynamicconfig.Key]any{
	  dynamicconfig.HistoryScannerEnabled: false,
	  dynamicconfig.HistoryScannerEnabled: false,
	  ...,
	})
```

Reply:
I support the idea of having CloneMapNonNil. At the same time I think setting field one by one is more readable. Alternatively, the could be a defaultOverwerrides map defined and then copied into clusterConfig.DynamicConfigOverrides which also will have reading easier:
```
defaultOverrides := map[dynamicconfig.Key]any{
   dynamicconfig.HistoryScannerEnabled: false,
   ...
}
maps.Copy(clusterConfig.DynamicConfigOverrides, defaultOverrides)
``` 
Event better will be to create a helper function which takes dynamicConfigOverrides and defaultOverrides and returns a new map:
```
clusterConfig.DynamicConfigOverrides := withDefaultOverrides(s.dynamicConfigOverrides, defaultOverrides)
```
and handle if nil then create inside the `withDefaultOverrides` function.